src/extractor/guidance/intersection_handler.cpp

```diff
@@ -400,108 +393,105 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
-    const auto sort_by = [](const out_way &lhs, const out_way &rhs) {
-        auto left_tie = std::tie(
-            rhs.deviation_from_straight, lhs.same_classification, lhs.same_or_higher_priority);
-        auto right_tie = std::tie(
-            lhs.deviation_from_straight, rhs.same_classification, rhs.same_or_higher_priority);
-        return left_tie > right_tie;
+    const auto SameName = [&](const EdgeData &way_data) {
+        return way_data.name_id == in_way.name_id;
+    };
+    auto sameOrHigherPriority = [&](const auto &way_data) {
+        return way_data.road_classification.GetPriority() <=
+               in_way.road_classification.GetPriority();
+    };
+    auto IsLowPriority = [&](const auto &way_data) {
+        return way_data.road_classification.IsLowPriorityRoadClass();
+    };
+    const auto RoadCompare = [&](const ConnectedRoad &lhs, const ConnectedRoad &rhs) {
+        const EdgeData &lhs_data = node_based_graph.GetEdgeData(lhs.eid);
+        const EdgeData &rhs_data = node_based_graph.GetEdgeData(rhs.eid);
+        const auto lhs_deviation = angularDeviation(lhs.angle, STRAIGHT_ANGLE);
+        const auto rhs_deviation = angularDeviation(rhs.angle, STRAIGHT_ANGLE);
+
+        bool rhs_same_or_priority = sameOrHigherPriority(rhs_data);
+        bool rhs_low_priority = IsLowPriority(rhs_data);
+        bool lhs_same_or_priority = sameOrHigherPriority(lhs_data);
+        bool lhs_low_priority = IsLowPriority(lhs_data);
+        auto left_tie = std::tie(rhs.entry_allowed, rhs_same_or_priority, lhs_low_priority, lhs_deviation);
+        auto right_tie = std::tie(lhs.entry_allowed, lhs_same_or_priority, rhs_low_priority, rhs_deviation);
+        return left_tie < right_tie;
     };
-    std::stable_sort(begin(out_ways), End, sort_by);
 
     // the ideal obvious turn candidate is the first way sorted to have the smallest deviation from
     // straight and is same or higher priority class
-    const auto sameOrHigher = [&](const out_way &way) {
-        return way.same_or_higher_priority == true;
-    };
-    auto best_candidate_it = std::find_if(begin(out_ways), End, sameOrHigher);
+    auto best_it = std::min_element(begin(intersection), end(intersection), RoadCompare);
 
-    // if no candidate is found with same or higher priority, use first non-low priority class
-    const auto notLowPriority = [&](const out_way &way) {
-        return !node_based_graph.GetEdgeData(way.road->eid)
-                    .road_classification.IsLowPriorityRoadClass();
-    };
-    if (best_candidate_it == End)
-        best_candidate_it = std::find_if(begin(out_ways), End, notLowPriority);
-
-    if (best_candidate_it != End)
+    BOOST_ASSERT(best_it != end(intersection));
+    best = std::distance(begin(intersection), best_it);
+    best_deviation = angularDeviation(intersection[best].angle, STRAIGHT_ANGLE);
+    const auto &best_data = node_based_graph.GetEdgeData(intersection[best].eid);
+    if (IsLowPriority(best_data))
```

Comment:
this check could be done right after the min_element, possibly combining both into a dedicated template tool function and making `best_it` const?

``` 
(pseuodo code)
template<iterator_type,comparator,filter>
iterator_type minimum_valid_element(iterator_type begin,iterator_type end,comparator comp, filter fil)
{
     const auto res = std::min_element(begin,end,comp);
     return fil(*res) ? end : fil;
};
```

Also: eliminating `low_priority` completely, what would you do in any of the normal turn scenarios, where all roads are service.

Needs some additional test cases with service.